contracts/upgradeable_contracts/amb_erc677_to_erc677/HomeStakeTokenMediator.sol

```diff
@@ -141,17 +140,16 @@ contract HomeStakeTokenMediator is BasicStakeTokenMediator, HomeStakeTokenFeeMan
             // burn all incoming tokens
             IBurnableMintableERC677Token(_token).burn(_value);
 
-            if (address(_blockRewardContract()) == address(0)) {
-                // in case if block reward contract is not configured, the fee is not collected
-                passMessage(_from, chooseReceiver(_from, _data), _value);
-            } else {
-                // when block reward contract is defined, the calculated fee is subtracted from the original value
+            if (isFeeCollectingActivated()) {
                 uint256 fee = calculateFee(_value);
+                // the calculated fee is subtracted from the original value
                 passMessage(_from, chooseReceiver(_from, _data), _value.sub(fee));
                 if (fee > 0) {
-                    // the fee itself is distributed later in the block reward contract
-                    _blockRewardContract().addBridgeTokenRewardReceivers(fee);
+                    // the fee manager will take care about fee distribution
+                    _distributeFee(fee);
```

Comment:
Do you think it makes sense to add events when fees are distributed from the mediators? In other bridges contracts we have:
https://github.com/poanetwork/tokenbridge-contracts/blob/165d828437826a43166ca787f8d39fb81f00c398/contracts/upgradeable_contracts/RewardableBridge.sol#L8-L9

They are useful to calculate the amount of fee distributed to display that information on the statistic page of the Bridge UI.
Although, for the specific case of STAKE token the event `BridgeTokenRewardAdded` from the block reward can be used for that purpose
 https://github.com/poanetwork/posdao-contracts/blob/c14fddb6bc9863f5d2c2075df3d755eb7637c712/contracts/base/BlockRewardAuRaTokens.sol#L74

Reply:
Yeah, I saw this event that's why decided to not include another one here. Will it complicate the logic of TokenBridge UI?

Reply:
No, it's OK to use that event in the Bridge UI. I just wanted to check that we considered the option and that using `BridgeTokenRewardAdded` is the correct way to go